File: 프로그래머스/0/120847. 최댓값 만들기 （1）/최댓값 만들기 （1）.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> numbers) {
    /* 조합을 이용한 풀이
    int ret = 0;
    for(int i = 0; i < numbers.size(); i++) {
        for(int j = i + 1; j < numbers.size(); j++) {
            ret = max(ret, numbers[i] * numbers[j]);
        }
    }
    return ret;
    
    */
    
    // sort (내림차순) 활용 풀이
    sort(numbers.begin(), numbers.end(), greater<int>());
    return numbers[0] * numbers[1];
}

File: 프로그래머스/0/120847. 최댓값 만들기 （1）/최댓값 만들기 （1）.swift
import Foundation

func solution(_ numbers:[Int]) -> Int {
    var result = 0
    for i in (0..<numbers.count) {
        for j in (i + 1..<numbers.count) {
            result = max(result, numbers[i] * numbers[j])
        }
    }
    return result
}